include/boost/fusion/container/set/set.hpp

```diff
@@ -97,39 +97,20 @@ namespace boost { namespace fusion
 
         template <typename Sequence>
         BOOST_FUSION_GPU_ENABLED
-        set(Sequence const& rhs)
-            : data(rhs) {}
-
-#if !defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
-        template <typename Sequence>
-        BOOST_FUSION_GPU_ENABLED
         set(Sequence&& rhs)
             : data(std::forward<Sequence>(rhs)) {}
-#endif
 
         BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         explicit
         set(typename detail::call_param<T>::type ...args)
             : data(args...) {}
 
-#if !defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
         template <typename ...U>
         BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         explicit
         set(U&& ...args)
```

Comment:
The implicit conversion constructor does not currently have the `constexpr` label. Is that intentional?

And I don't see how the referenced constexpr evaluation case applies to SFINAE in this situation. There is no reason to use a constexpr function in the SFINAE expression I proposed. Maybe I'm missing something.

That constructor (`template<typename Sequence> set(Sequence&&)`) should likely be marked explicit if SFINAE cannot be used for some reason. I think it would be nice to keep existing behavior, which is implicitly convertible from any fusion sequence type and explicitly constructible from a matching list of items, but this obviously isn't my decision.